src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -775,7 +775,38 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
 
                                 // newName == nullptr likely means we have an empty search string.  We should leave newNameToUse
                                 // as nullptr so we clear the renamed column
-                                if (newName != nullptr)
+                                // Except string transformation is selected.
+                                
+                                if (newName == nullptr && (flags & Uppercase || flags & Lowercase || flags & Titlecase))
+                                {
+                                    newName = new wchar_t[MAX_PATH]{ 0 };
+                                    if (flags & NameOnly)
+                                    {
+                                        StringCchCopy(newName, MAX_PATH, fs::path(originalName).stem().c_str());
+                                    }
+                                    else if (flags & ExtensionOnly)
+                                    {
+                                        std::wstring extension = fs::path(originalName).extension().wstring();
+                                        if (!extension.empty())
+                                        {
+                                            if (extension.front() == '.')
+                                            {
+                                                extension = extension.erase(0, 1);
```

Comment:
No need to do this assignment, erase works in place

Reply:
Exact block of code was already present in the code, so I just wanted to follow the pattern. 
We can change both to `extension.erase(0, 1)`

Reply:
In that case no need to change it